test/kubernetes/e2e/features/services/tcproute/testdata/single-backend-service.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: single-svc
+  namespace: single-tcp-route
+  labels:
+    app: single-svc
+spec:
+  ports:
+    - name: http
+      port: 3001
+      targetPort: 3000
+  selector:
+    app: backend-0
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: backend-0
+  namespace: single-tcp-route
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: backend-0
+      version: v1
+  template:
+    metadata:
+      labels:
+        app: backend-0
+        version: v1
+    spec:
+      containers:
+        - image: gcr.io/k8s-staging-gateway-api/echo-basic:v20231214-v1.0.0-140-gf544a46e
```

Comment:
nit: if we can link to upstream so that we stay in lock step that would be cool. 

That being said this is an echo server that seems fairly stable as it goes all the way to ingress api and likely doesnt need much changing.